### PR TITLE
Add the settings to set default thumbnail for video and audio files.

### PR DESCRIPTION
**Pull request by dharmin**
Solved the conflict from original PR : #1037

### CODE REVIEW

File: app/admin/RTMediaFormHandler.php [hunk 1]
```diff
@@ -348,6 +348,37 @@ public static function fileinput( $args ) {
 		$fileObj->display_file_input( $args );
 	}
 
+	/**
+	 * Show rtmedia input file in admin options.
+	 *
+	 * @access static
+	 *
+	 * @param  array $args arguments to create file input control for default thumbnail generator settings.
```
Comment (SID177) [inline app/admin/RTMediaFormHandler.php]: Keep the first word Camelcase.

File: app/admin/RTMediaFormHandler.php [hunk 2]
```diff
@@ -348,6 +348,37 @@ public static function fileinput( $args ) {
...
+	 *
+	 * @return void
+	 */
+	public static function inputfile( $args ) {
+		global $rtmedia;
+		$options  = $rtmedia->options;
```
Comment (SID177) [inline app/admin/RTMediaFormHandler.php]: Alignment. Please run phpcs once.

File: app/helper/rtForm.php [hunk 3]
```diff
@@ -579,6 +583,61 @@ protected function generate_textbox( $attributes ) {
 			}
 		}
 
+		/**
+		 * Generate rtmedia html inputfile in admin options.
+		 *
+		 * @access protected
+		 *
+		 * @param $attributes arguments to create file input control for default thumbnail generator settings.
```
Comment (SID177) [inline app/helper/rtForm.php]: Keep first word camelcase. Also add parameter type.

File: app/helper/rtForm.php [hunk 4]
```diff
@@ -661,6 +720,30 @@ public function get_link( $attributes = '' ) {
 		public function display_link( $args = '' ) {
 			echo $this->get_link( $args );
 		}
+		/**
+		 * Get rtmedia html input file in admin options.
+		 *
+		 * @access public
+		 *
+		 * @param string|array $attributes
```
Comment (SID177) [inline app/helper/rtForm.php]: Please add parameter description

File: app/helper/rtForm.php [hunk 5]
```diff
@@ -661,6 +720,30 @@ public function get_link( $attributes = '' ) {
...
+		 */
+		public function get_inputfile( $attributes = '' ) {
+			return $this->generate_inputfile( $attributes );
+		}
+
+
+        /**
+         * Display file input settings for set custom thumbnail generator.
+         *
+         * @param $args
```
Comment (SID177) [inline app/helper/rtForm.php]: Please add parameter description and type.

File: app/helper/rtForm.php [hunk 6]
```diff
@@ -661,6 +720,30 @@ public function get_link( $attributes = '' ) {
 		public function display_link( $args = '' ) {
 			echo $this->get_link( $args );
 		}
+		/**
```
Comment (SID177) [inline app/helper/rtForm.php]: Leave one line above this.

File: app/helper/rtForm.php [hunk 7]
```diff
@@ -661,6 +720,30 @@ public function get_link( $attributes = '' ) {
...
+		}
+
+
+        /**
+         * Display file input settings for set custom thumbnail generator.
+         *
+         * @param $args
+         *
+         * @throws rtFormInvalidArgumentsException
+         */
```
Comment (SID177) [inline app/helper/rtForm.php]: Alignment.

File: app/helper/rtForm.php [hunk 8]
```diff
@@ -662,6 +722,31 @@ public function display_link( $args = '' ) {
...
+
+
+		/**
+		 * Display file input settings for set custom thumbnail generator.
+		 *
+		 * @param array $args Arguments to create file input control.
+		 *
+		 * @throws rtFormInvalidArgumentsException Invalid argument exception.
+		 */
+		public function display_inputfile( $args = '' ) {
```
Comment (SID177) [inline app/helper/rtForm.php]: There is array in parameter description but empty string is assigned to `$args`. You can change parameter type to `array|string`.

File: app/helper/rtForm.php [hunk 9]
```diff
@@ -243,20 +246,21 @@ private function generate_element_value( $element, $attributes ) {
 				case 'rtNumber': //number
 				case 'rtText' : //text
 				case 'rtButton': //button
+				case 'rtFile': //file
```
Comment (milindmore22) [inline app/helper/rtForm.php]: the inline comment must end with full stop need a space between `comment` and `//`

File: app/helper/rtForm.php [hunk 10]
```diff
@@ -579,6 +583,62 @@ protected function generate_textbox( $attributes ) {
...
+			if ( is_array( $attributes ) ) {
+
+				/* Starting the input tag */
+				$html = '<input type="file" ';
+
+				/* Generating attributes */
+				$html .= $this->processAttributes( $element, $attributes );
+
+				/* Ending the tag */
+				$html .= ' /><input type="hidden" name="rtmedia-options[' . esc_attr( $attributes['name'] ) . '_hid]" value="' . esc_attr( $attributes['value'] ) . '" />';
```
Comment (milindmore22) [inline app/helper/rtForm.php]: validate `$attributes['name']` and `$attributes['value']`  before using them dont trust inputs.

File: app/helper/rtForm.php [hunk 11]
```diff
@@ -662,6 +722,31 @@ public function display_link( $args = '' ) {
...
+
+		/**
+		 * Display file input settings for set custom thumbnail generator.
+		 *
+		 * @param string|array $args Arguments to create file input control.
+		 *
+		 * @throws rtFormInvalidArgumentsException Invalid argument exception.
+		 */
+		public function display_inputfile( $args = '' ) {
+			echo $this->get_inputfile( $args );
```
Comment (milindmore22) [inline app/helper/rtForm.php]: add comment above `Previously escaped and sanitized so not required here.`
add a comment on this line `// WPCS: XSS ok.`

**Pull request merged**